### PR TITLE
Add existingResourcePolicy restore CR validation to controller.

### LINKED ISSUE #7347: `existingResourcePolicy` validation outside CLI
**What steps did you take and what happened:**
Created a restore using kubectl command and set existingResourcePolicy to "batman". Restore succeed without mentioned wrong existingResourcePolicy.

**What did you expect to happen:**
Warn user that is not a valid existingResourcePolicy and present user with available options.

Validation present in CLI command
https://github.com/vmware-tanzu/velero/blob/270b1de6a1fbfd672b2c88f46ed1ffbbdfac864c/pkg/cmd/cli/restore/create.go#L424-L429

**Anything else you would like to add:**
As previously discussed, we do not want to add enum validation in CRD, as there are plans to add more options to that field in the future https://github.com/vmware-tanzu/velero/issues/6142 

So, its best to add validation through code

**Environment:**

- Velero version (use `velero version`): 1.12

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "I would like to see this bug fixed as soon as possible"
- :-1: for "There are more important bugs to focus on right now"

### PR DESCRIPTION
**Pull request by kaovilai**
Signed-off-by: Tiger Kaovilai <tkaovila@redhat.com>

Thank you for contributing to Velero!

# Please add a summary of your change

# Does your change fix a particular issue?

Fixes #7347 

# Please indicate you've done the following:

- [x] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [ ] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

**Review by sseago — APPROVED** [review]

**Review by anshulahuja98 — DISMISSED** [review]

**Comment by anshulahuja98** [comment]
@kaovilai CI is failing, can you pl check
Verifying goimports - failed! Please run 'make update'.

**Comment by kaovilai** [comment]
Fixed `make update`! Thanks all.

**Review by sseago — DISMISSED** [review]

**Review by anshulahuja98 — DISMISSED** [review]

File: pkg/restore/util/util.go [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package util
```
Comment (blackpiglet) [inline pkg/restore/util/util.go:1]: It's better to move this file into `pkg/util/` directory.

Comment (reasonerjt) [reply pkg/restore/util/util.go:1]: +1

**Comment by kaovilai** [comment]
labels PR fix https://github.com/vmware-tanzu/velero/pull/7776

**Review by blackpiglet — APPROVED** [review]

**Pull request merged**